crates/forge/bin/cmd/build.rs

```diff
@@ -80,7 +81,22 @@ impl BuildArgs {
 
         let project = config.project()?;
 
+        // Collect sources to compile if build subdirectories specified.
+        let mut files = vec![];
+        if let Some(dirs) = self.args.dirs {
+            for dir in dirs {
+                for entry in fs::read_dir(dir)? {
+                    let entry = entry?;
+                    let path = entry.path();
+                    if path.is_file() {
+                        files.push(entry.path());
+                    }
+                }
+            }
+        }
```

Comment:
I believe there should be utils to find the sol/vy files in a dir @klkvr ?

I believe
https://github.com/foundry-rs/compilers/blob/43da903df7231bf60377c85a6d76ae6b088c723d/src/utils.rs#L113-L113

Reply:
Right, similarly to this https://github.com/foundry-rs/foundry/blob/b7dcf4661ddb4fa0cf3519c800df2d5c96500a7e/crates/forge/bin/cmd/test/mod.rs#L219

Reply:
nice, changed it in https://github.com/foundry-rs/foundry/pull/8149/commits/6da9a55dfb23d7202ff0018d8441f19fe1c37efc